pkg/pget.go

```diff
@@ -39,26 +39,27 @@ func (g *Getter) DownloadFile(ctx context.Context, url string, dest string) (int
 	if err != nil {
 		return fileSize, 0, err
 	}
-	downloadElapsed := time.Since(downloadStartTime)
-	writeStartTime := time.Now()
+	// downloadElapsed := time.Since(downloadStartTime)
+	// writeStartTime := time.Now()
 
 	err = g.Consumer.Consume(buffer, dest)
 	if err != nil {
 		return fileSize, 0, fmt.Errorf("error writing file: %w", err)
 	}
-	writeElapsed := time.Since(writeStartTime)
+	// writeElapsed := time.Since(writeStartTime)
 	totalElapsed := time.Since(downloadStartTime)
 
 	size := humanize.Bytes(uint64(fileSize))
-	downloadThroughput := humanize.Bytes(uint64(float64(fileSize) / downloadElapsed.Seconds()))
-	writeThroughput := humanize.Bytes(uint64(float64(fileSize) / writeElapsed.Seconds()))
+	// downloadThroughput := humanize.Bytes(uint64(float64(fileSize) / downloadElapsed.Seconds()))
+	// writeThroughput := humanize.Bytes(uint64(float64(fileSize) / writeElapsed.Seconds()))
 	logger.Info().
 		Str("dest", dest).
+		Str("url", url).
 		Str("size", size).
-		Str("download_throughput", fmt.Sprintf("%s/s", downloadThroughput)).
-		Str("download_elapsed", fmt.Sprintf("%.3fs", downloadElapsed.Seconds())).
-		Str("write_throughput", fmt.Sprintf("%s/s", writeThroughput)).
-		Str("write_elapsed", fmt.Sprintf("%.3fs", writeElapsed.Seconds())).
+		// Str("download_throughput", fmt.Sprintf("%s/s", downloadThroughput)).
+		// Str("download_elapsed", fmt.Sprintf("%.3fs", downloadElapsed.Seconds())).
+		// Str("write_throughput", fmt.Sprintf("%s/s", writeThroughput)).
+		// Str("write_elapsed", fmt.Sprintf("%.3fs", writeElapsed.Seconds())).
```

Comment:
We lost the overall throughput and such for the single file mode. Maybe we can push that data back up to cmd.root to look the same as multifile with `[ metrics ]` as the .Msg()?

Reply:
(we had this conversation IRL but just jotting down my position for the peanut gallery)

I think we need to be very specific about what we're trying to measure and why. Given we download individual files in parallel and during download we might have arbitrary blocking on the semaphore, what measurement do we want to make to indicate how fast an individual file was downloaded?

Similar questions apply to the writeThroughput. Given we now interleave writing with downloading, what specifically can/should we measure to show write throughput?

I do understand that it'd be good to identify where any bottlenecks are but I don't think there's a meaningful way we can say how fast a given file was downloaded. We can maybe do something for downloading chunks, but we wouldn't want a log line for every single chunk. Maybe we collect per-chunk download statistics (mean, P05/P50/P95 size/duration/throughput) and publish them? Possibly by upstream host?

Reply:
I'd also say that `download_elapsed` is broken in v0.4.1 already.